### ISSUE TITLE
Add option to not generate fake data

### ISSUE THREAD

**Issue by jchannon** [jfloodnet/Csharp2Json#1]
Hi,

Would it be possible to add a checkbox to not generate fake data so you get the class serialized but with empty values?

**Comment by jfloodnet** [comment]
I had a feeling this would be your request :) Which types would you like to apply this too? Strings only? 

This is simple enough to do for strings I believe, have a read of http://blog.ploeh.dk/2009/04/27/ReplacingAutoFixturesDefaultAlgorithms/

**Comment by jchannon** [comment]
I was going to say all of them but just looked at the code so not sure what will work.

If you returned `return MakeTypes.From(source);` would that do the trick?

**Comment by jfloodnet** [comment]
Well not really, MakeTypes literally makes an array of "Type", you would still need an instance for serialisation. AutoFixture is what generates the instances, so you want to customise the data it injects when creating instances. 

You could configure AutoFixture to return "default(T)" for every property, you would then get null for strings and 0 for ints. The problem with this approach is it will also generate null for any nested type you have defined. 

I'm assuming what you want is a fully defined Json structure covering all the types defined, but with default values for primitive types?

**Comment by jchannon** [comment]
yeah that or just empty strings for each property

**Comment by jfloodnet** [comment]
So inside, MakeObjects - something like this would do the trick:

```
        var fixture = new Fixture();
        if (!withFakeData)
        {
            fixture.Register(() => "");
        }
        var context = new SpecimenContext(fixture);
        foreach (var item in types)
            yield return fixture.Create(item, context);
```

But you still get random values for Guid / int / Enum / bool etc -

**Comment by jchannon** [comment]
I'd still want those types to empty too

**Comment by jfloodnet** [comment]
Yep - so you could carry on the same way registering an empty value for each of the built in types: 

i.e. fixture.Register(() => Guid.NewGuid()); fixture.Register(() => 0) etc

OR you can look into creating a custom ISpecimenBuilder that checks if the request is for a primitive type and generates an "empty" value suitable for that type.

This article is not bad to get you started:

http://thomasardal.com/customizations-for-autofixture-my-new-best-friend/
